File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Thumbnail.java

package com.noblemajesty.marvel.models.getCharacters;

import android.os.Parcel;
import android.os.Parcelable;
import android.os.Parcelable.Creator;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Thumbnail implements Parcelable
{

  @SerializedName("path")
  @Expose
  private String path;
  @SerializedName("extension")
  @Expose
  private String extension;
  public final static Parcelable.Creator<Thumbnail> CREATOR = new Creator<Thumbnail>() {


    @SuppressWarnings({
        "unchecked"
    })
    public Thumbnail createFromParcel(Parcel in) {
      return new Thumbnail(in);
    }

    public Thumbnail[] newArray(int size) {
      return (new Thumbnail[size]);
    }

  };

  protected Thumbnail(Parcel in) {
    this.path = ((String) in.readValue((String.class.getClassLoader())));
    this.extension = ((String) in.readValue((String.class.getClassLoader())));
  }

  public Thumbnail() {
  }

  public String getPath() {
    return path;
  }

  public String getExtension() {
    return extension;
  }


  public void writeToParcel(Parcel dest, int flags) {
    dest.writeValue(path);
    dest.writeValue(extension);
  }

  public int describeContents() {
    return 0;
  }

}

File: app/src/main/java/com/noblemajesty/marvel/models/getCharacters/Thumbnail.kt
package com.noblemajesty.marvel.models.getCharacters

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Thumbnail : Parcelable {

    @SerializedName("path")
    @Expose
    val path: String
    @SerializedName("extension")
    @Expose
    val extension: String

    protected constructor(`in`: Parcel) {
        this.path = `in`.readValue(String::class.java.classLoader) as String
        this.extension = `in`.readValue(String::class.java.classLoader) as String
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(path)
        dest.writeValue(extension)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        val CREATOR: Parcelable.Creator<Thumbnail> = object : Creator<Thumbnail> {


            override fun createFromParcel(`in`: Parcel): Thumbnail {
                return Thumbnail(`in`)
            }

            override fun newArray(size: Int): Array<Thumbnail?> {
                return arrayOfNulls(size)
            }

        }
    }

}